Improve readme w/ better phantomjs & .env instructions

craiglyons opened the pull request:
- [x] Clarify `phantomjs` install
- [x] Clarify `.env` config & server running instructions
- [x] Improve markdown
- [x] Update terminal instructions

nfiniteset commented on README.md:
```diff
@@ -21,59 +22,95 @@ What's in the box?
+
+- `heroku local` will read your `.env` and start the server locally, and has the benefit of similarities to your production environment.
+- `foreman` is another good option.
+- You can also do this on your own with `source .env` and `mix phoenix.server`
```
I wonder if there's a reasonable default to recommend here rather than options.

craiglyons replied:
Good idea.  Per our chat, we should change the instructions to inline the env vars, and just quickly suggest the foreman / heroku local alternatives.

craiglyons commented:
@nfiniteset I switched over the `.env` stuff. Would you mind spinning up a sample app with the new instructions & see if your issues are covered?  I updated both the readme & the terminal instructions.
